inductiva/projects/project.py

```diff
@@ -0,0 +1,266 @@
+        self.id = model.get("id")
+
+        return self
+
+    def open(self):
+        """Open the project.
+
+        Open the project and make it the active one in the calling
+        thread. An opened project will ensure that calls to the
+        `get_current_project` function will return this project.
+        Consecutive calls to this method are idempotent.
+
+        Raises:
+            RuntimeError if another opened project exists, i.e.,
+            `get_current_project` returns something other than None.
+
+        """
+        _logger.debug("Opening project %s", self.name)
+        current_project = get_current_project()
+
+        if current_project is self:
+            _logger.debug("Project is already opened.")
+            return
+
+        if current_project is not None:
+            raise RuntimeError(
+                "Trying to open a project when another is active.")
+
+        self._token = _CURRENT_PROJECT.set(self)
+
+    def close(self):
+        """Close the project.
+
+        Calls to the `get_current_project` will return `None` after
+        the project is closed.  Consecutive calls to this method are
+        idempotent.
+
+        """
+        if self._token is None:
+            return
+
+        _CURRENT_PROJECT.reset(self._token)
+        self._token = None
+
+    @property
+    def opened(self) -> bool:
+        """Checks if the project is open."""
+        return self._token is not None
+
+    @property
+    def info(self) -> ProjectInfo:
+        """Returns project info.
+
+        It makes no requests to the backend. Instead it uses
+        previously stored information. Therefore it can be
+        outdated. For a method that requests the most recent data from
+        the backend use: `get_info`.
+
+        """
+        return self._info
+
+    def get_info(self) -> ProjectInfo:
+        """Get project information.
+
+        Get updated information on the project. This method executes a
+        call to the backend to retrieve the most recent information
+        about this project and stores it internally so that it becomes
+        available through the `Project.info` property.  This method is
+        suitable when up-to-date information is required.
+
+        """
+        name = self.name
+        model = self._get_model(name)
+        self._update_from_api_response(model)
+        return self._info
+
+    def __str__(self) -> str:
+        return f"Project '{self.name}' with "\
+               f"{self.num_tasks} tasks (id={self.id})"
+
+    def desc(self) -> str:
+        """Generates a string description of the object
+
+        Returns:
+          str: A string descrption of the object. Includes project name,
+            number of tasks and a description of the tasks' status.
+
+        """
+        header = str(self) + "\n"
+        summary = "\n".join(
+            f"  {k}: {v}" for k, v in self._info.task_by_status.items())
+        return header + summary
+
+    def get_tasks(self,
```

Comment:
how can we get all the tasks that belong to this project?
maybe if we call it with `last_n = -1`?

Reply:
I don't like this approach. It would make more sense to me that any `n <= 0` should return all tasks.